File: 人机交互2020/9-4/9-4/1-4 VB.net+树形类+应用/Test/clsDir.vb
﻿Imports System.IO

Public Class clsDir
    Private files() As String
    Private dirs() As clsDir

    Public Sub New(ByVal path As String)
        files = Directory.GetFiles(path)
        Dim folders() As String = Directory.GetDirectories(path)
        ReDim dirs(folders.Count - 1)
        For i = 0 To folders.Count - 1
            dirs(i) = New clsDir(folders(i))
        Next
    End Sub
    Public Sub Output(ByVal lstInfo As ListBox)
        For i = 0 To files.Count - 1
            lstInfo.Items.Add(files(i))
        Next
        For i = 0 To dirs.Count - 1
            dirs(i).Output(lstInfo)
        Next
    End Sub
    Public Function Depth() As Integer
        Return 0
    End Function
    Public Function Count() As Integer
        Return 0
    End Function
    Public Function Search(ByVal pattern As String) As List(Of String)
        Dim res As New List(Of String)
        Return res
    End Function

End Class

File: 人机交互2020/9-4/2-4 C#+树形类+应用/Test/clsDir.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.IO;
using System.Windows.Forms;

namespace Test
{
    class clsDir
    {
        private List<clsDir> Dirs;
        private String[] Files;
        public clsDir(String path)
        {
            String []folders= Directory.GetDirectories(path);
            Dirs=new  List<clsDir>();
            for (int i = 0; i < folders.Count(); i++)
                Dirs.Add(new clsDir(folders[i]));
            Files = Directory.GetFiles(path);
        }
        public void  Output(ListBox lstInfo)
        {
            for (int i = 0; i < Dirs.Count(); i++)
                Dirs[i].Output(lstInfo) ;
            for (int i = 0; i < Files.Count(); i++)
                lstInfo.Items.Add(Files[i]);
        }
        public int Depth()
        {
            return 0;
        }
        public int Count()
        {
            return 0;
        }
        public List<String> Search(String patter)
        {
            List<String> res = new List<String>();
            return res;
        }
    }
}
